### PR TITLE
Issue #577: Removing node delete operations from admin nav.

### LINKED ISSUE #577: [UX] Remove the "Delete" links for content types from admin bar
It worries me that the top item in the child drop-down for content types is "Delete".  This puts undue importance on that link.

On further investigation, there isn't even a menu link for Delete. Let's remove these.

### PR DESCRIPTION
**Pull request by serundeputy**
This fixes: https://github.com/backdrop/backdrop-issues/issues/577

### CODE REVIEW

**Comment by quicksketch** [comment]
I think this will prevent the delete page from being accessed at all(?)

**Comment by serundeputy** [comment]
yep; you are correct without that `$items` entry you can not delete the content types.  oops :(

i'll look again.

File: core/modules/admin_bar/admin_bar.inc [hunk 1]
```diff
@@ -225,14 +225,16 @@ function admin_bar_tree_dynamic(array $expand_map) {
...
     // provide the expansion map arguments to apply to the dynamic tree.
-    if (isset($expand_map[$link['path']])) {
-      $parent_path = $expand_map[$link['path']]['parent'];
-      $link['expand_map'] = $expand_map[$link['path']]['arguments'];
-    }
-    // Otherwise, just store this link keyed by its parent path; the expand_map
-    // is automatically derived from parent paths.
-    else {
-      $parent_path = $result[$link['plid']]['path'];
+    if ($mlid != 278) {
```
Comment (serundeputy) [inline core/modules/admin_bar/admin_bar.inc]: @quicksketch can we rely on this 278 from mlid column of the menu_links db table?

Comment (quicksketch) [inline core/modules/admin_bar/admin_bar.inc]: No I don't think so. the mlid will change from site to site. Is `$link['type']` available here?

Comment (serundeputy) [inline core/modules/admin_bar/admin_bar.inc]: `$link['type']` is available, but is `int`s looks like could be `mlid`s or `plid`s not sure

Comment (serundeputy) [inline core/modules/admin_bar/admin_bar.inc]: can we rely on: 
`if ($link['router_path'] != 'admin/structure/types/manage/%/delete') {` 
?

Comment (quicksketch) [inline core/modules/admin_bar/admin_bar.inc]: I think what we should do is two-fold:
- Update `'admin/structure/types/manage/%node_type/delete'` in `node_menu` to add: `'type' => MENU_CALLBACK,`
- Then check if `$link['type'] != MENU_CALLBACK` here. (MENU_CALLBACK should have the value of "0").

Comment (serundeputy) [inline core/modules/admin_bar/admin_bar.inc]: that works!  ... i'll rebase and commit.  thanks!

**Comment by quicksketch** [comment]
Cherry-picked back into 1.0.x as well.

**Pull request merged**